generator/src/main/java/org/corfudb/generator/operations/NestedTxOperation.java

```diff
@@ -0,0 +1,55 @@
+package org.corfudb.generator.operations;
+
+import org.corfudb.generator.State;
+import org.corfudb.runtime.exceptions.TransactionAbortedException;
+
+import java.util.List;
+
+/**
+ * Created by rmichoud on 7/26/17.
+ */
+public class NestedTxOperation extends Operation {
+
+    private final int maxNest = 20;
+    public NestedTxOperation(State state) {
+        super(state);
+    }
+
+    @Override
+    public void execute() {
+        state.startOptimisticTx();
+
+
+
+        int numNested = state.getOperationCount().sample(1).get(0);
+        int nestedTxToStop = numNested;
+        for (int i = 0; i < numNested && i < maxNest; i++) {
+            try{
+            state.startOptimisticTx();
+            int numOperations = state.getOperationCount().sample(1).get(0);
+            List<Operation> operations = state.getOperations().sample(numOperations);
+
+            System.out.println("Start nested Tx of " + numOperations + "operations");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)